src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs

```diff
@@ -120,22 +124,26 @@ public class AadIssuerValidator
 
             try
             {
-                if (securityToken.Issuer.EndsWith("v2.0", StringComparison.OrdinalIgnoreCase))
+                string AadIssuer;
+                if (validationParameters.ValidateIssuerWithLKG)
                 {
-                    if (AadIssuerV2 == null)
-                        AadIssuerV2 = CreateConfigManager(AadAuthorityV2).GetConfigurationAsync().ConfigureAwait(false).GetAwaiter().GetResult().Issuer;
-
-                    if (IsValidIssuer(AadIssuerV2, tenantId, issuer))
-                        return issuer;
+                    AadIssuer = validationParameters.ConfigurationManager.LastKnownGoodConfiguration.Issuer;
```

Comment:
We can't change the logic here to use the ConfigurationManager in TokenValidationParameters. The issueer is obtained the first time a token is validated. The logic is to use the v1 or v2 endpoint and then adjust for common, which has a '{tenant}' value in the 'issuer' string.

It may make sense to have LKG and Refresh logic in AadIssuerValidator.
The flag you added to TokenValidationParameters.ValidateIssuerWithLKG, would signal to AadIssuerValidator to use its LKG value.
We will also need a way to instruct AadIssuerValidator to refresh, this should be throttled just as ConfigurationManager is.
Currently AadIssuerValidator does not have a cache that limits entries so caching ConfigurationManagers would need to use the LRUCache.

Reply:
LKG also need to be for both v1 and v2. Thinking what if GetConfigurationAsync gets both the v1 and v2 config's every time and we can have additional api's Getconfigv1 and Getconfigv2 that call GetConfigurationAsync internally and return respective config. That way refresh only after certain time logic also applies and the TVP.configmanager is the one that SAL fetch from cache.

Reply:
AadIssuerValidator can be used independently without SAL as well.

Reply:
After thinking more about it, it makes sense for AadIssuerValidator to have LKGv1/v2 of its own. If issuer is invalid it could try respective LKG first and later try respective configmanagerv1/v2 and pull in fresh config.